src/components/expandable-row.vue

```diff
@@ -0,0 +1,76 @@
+<!--
+Copyright 2023 ODK Central Developers
+See the NOTICE file at the top-level directory of this distribution and at
+https://github.com/getodk/central-frontend/blob/master/NOTICE.
+
+This file is part of ODK Central. It is subject to the license terms in
+the LICENSE file found in the top-level directory of this distribution and at
+https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+-->
+<template>
+  <tr class="expandable-row">
```

Comment:
It'd be great to add unit tests of this component that verify that the component toggles correctly and uses the provided slots. That might have the result of simplifying other component tests: in those other tests, you will be able to just test that this component has been passed the correct content, not anything about the toggling.